tests/middlewareTypes.test.tsx

```diff
@@ -72,7 +34,7 @@ describe('counter state spec (no middleware)', () => {
 
 describe('counter state spec (single middleware)', () => {
   it('immer', () => {
-    const useStore = create<CounterState>(
+    const useStore = create<CounterState>()(
```

Comment:
Yeah, this is truly magical, I wonder if people would like it or not.
In JS, we don't need this right?
What if people want the v3 style, with explicit typing, would it be possible? I'd like to have such an option too.

Reply:
The type seems to support it. Can you add at least one example in this file please?
Maybe, subscribeWithSelector + persist combo would be a good example.

Reply:
> Yeah, this is truly magical, I wonder if people would like it or not.

I think TS people would like it as they don't have to manually pass the complex middlewared store types.

> In JS, we don't need this right?

Yes, although it would be preferred for better intellisense.

> What if people want the v3 style, with explicit typing, would it be possible? I'd like to have such an option too.
> The type seems to support it. Can you add at least one example in this file please?
> Maybe, subscribeWithSelector + persist combo would be a good example.

Yep it's possible I'll add a test case